kornia/geometry/transform/crop.py

```diff
@@ -201,29 +208,67 @@ def _infer_bounding_box(boxes: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor
           coordinates must be in the x, y order.
 
     Returns:
-        (torch.Tensor, torch.Tensor): tensor containing the patches with shape BxN1xN2
+        Tuple[torch.Tensor, torch.Tensor]:
+        - Bounding box heights, shape of :math:`(B,)`.
+        - Boundingbox widths, shape of :math:`(B,)`.
 
     Example:
         >>> boxes = torch.tensor([[
-                [1., 1.],
-                [2., 1.],
-                [2., 2.],
-                [1., 2.],
-            ]])  # 1x4x2
-        >>> _infer_bounding_box(boxes)
-        (2, 2)
+        ...     [1., 1.],
+        ...     [2., 1.],
+        ...     [2., 2.],
+        ...     [1., 2.],
+        ... ], [
+        ...     [1., 1.],
+        ...     [3., 1.],
+        ...     [3., 2.],
+        ...     [1., 2.],
+        ... ]])  # 2x4x2
+        >>> infer_box_shape(boxes)
+        (tensor([2., 2.]), tensor([2., 3.]))
     """
+    validate_bboxes(boxes)
+    width: torch.Tensor = (boxes[:, 1, 0] - boxes[:, 0, 0] + 1)
+    height: torch.Tensor = (boxes[:, 2, 1] - boxes[:, 0, 1] + 1)
+    return (height, width)
+
+
+def validate_bboxes(boxes: torch.Tensor) -> None:
     assert torch.allclose((boxes[:, 1, 0] - boxes[:, 0, 0] + 1), (boxes[:, 2, 0] - boxes[:, 3, 0] + 1)), \
         "Boxes must have be square, while get widths %s and %s" % \
         (str(boxes[:, 1, 0] - boxes[:, 0, 0] + 1), str(boxes[:, 2, 0] - boxes[:, 3, 0] + 1))
     assert torch.allclose((boxes[:, 2, 1] - boxes[:, 0, 1] + 1), (boxes[:, 3, 1] - boxes[:, 1, 1] + 1)), \
         "Boxes must have be square, while get heights %s and %s" % \
         (str(boxes[:, 2, 1] - boxes[:, 0, 1] + 1), str(boxes[:, 3, 1] - boxes[:, 1, 1] + 1))
-    assert len((boxes[:, 1, 0] - boxes[:, 0, 0] + 1).unique()) == 1, \
-        "Boxes can only have one widths, got %s" % str((boxes[:, 1, 0] - boxes[:, 0, 0] + 1).unique())
-    assert len((boxes[:, 2, 1] - boxes[:, 0, 1] + 1).unique()) == 1, \
-        "Boxes can only have one heights, got %s" % str((boxes[:, 2, 1] - boxes[:, 0, 1] + 1).unique())
 
-    width: torch.Tensor = (boxes[:, 1, 0] - boxes[:, 0, 0] + 1)[0]
-    height: torch.Tensor = (boxes[:, 2, 1] - boxes[:, 0, 1] + 1)[0]
-    return (height, width)
+
+def bbox_to_mask(boxes: torch.Tensor, width: int, height: int) -> torch.Tensor:
```

Comment:
this needs proper documentation. Not included to sphinx docs.

Reply:
Besides, I'm thinking since we start to have few functionalities around bounding boxes manipulation, we could have a small sub-module dedicated to that: `kornia.geometry.transform.boxes` ? /cc @shijianjian @ducha-aiki @lferraz 
I think @lferraz had some suggestions about this